yok.ts

```diff
@@ -137,9 +137,29 @@ export class Yok implements IInjector {
 		forEachName(names, (name) => this.requireOne(name, file));
 	}
 
+	public publicApi: any = {};
+	public _publicApi: any = {};
+
+	public requirePublic(names: any, file: string): void {
+		forEachName(names, (name) => {
+			this.requireOne(name, file);
+			this.resolvePublicApi(name, file);
+		});
+	}
+
+	private resolvePublicApi(name: string, file: string): void {
+		Object.defineProperty(this.publicApi, name, {
+			get: () => {
+				let obj: any = {};
+				let originalModule = this.resolve(name);
```

Comment:
Indeed, I saw this after I posted my comment. Having that in mind I would suggest to rename the `promisify` decorator to `public` or `export` to better donate what is is doing under the hood.